vtkext/private/module/vtkF3DPointSplatMapper.cxx

```diff
@@ -84,8 +104,167 @@ void vtkF3DSplatMapperHelper::BuildBufferObjects(vtkRenderer* ren, vtkActor* act
 
   vtkOpenGLPointGaussianMapperHelper::BuildBufferObjects(ren, act);
 
+  // allocate a buffer of depths used for sorting splats
   this->DepthBuffer->Allocate(splatCount * sizeof(float), vtkOpenGLBufferObject::ArrayBuffer,
     vtkOpenGLBufferObject::DynamicCopy);
+
+  this->SphericalHarmonicsDegree = 0;
+
+  // we store each spherical harmonics packed in a RGB 8-bits texture.
+  // we use the GPU maximum texture size to set the width of the texture.
+  // the height will depends on the number of gaussians.
+  glGetIntegerv(GL_MAX_TEXTURE_SIZE, &this->MaxTextureSize);
+
+  if (splatCount > this->MaxTextureSize * this->MaxTextureSize)
+  {
+    vtkWarningMacro("Gaussians count is too high to support spherical harmonics on this GPU");
+    return;
+  }
```

Comment:
may be reachable in a dedicated cxx test, as you see fit

Reply:
It's GPU dependent.
On my GPU `MaxTextureSize` is 32,768. So up to 1,073,741,824 gaussians are supported. I won't create a `vtkPoints` (and related attributes) with this amount of data.  
I think we should keep this test to avoid any crash on low-end GPU.

Reply:
Fair enough.

of course, lets keep the test.